perfkitbenchmarker/beam_benchmark_helper.py

```diff
@@ -137,7 +171,65 @@ def BuildMavenCommand(benchmark_spec, classname, job_arguments):
   cmd.append("-DintegrationTestPipelineOptions="
              "[{}]".format(','.join(beam_args)))
 
+  return cmd
+
+
+def _BuildPythonCommand(benchmark_spec, modulename, job_arguments):
+  """ Constructs a Python command for the benchmark.
+
+  Args:
+    benchmark_spec: The PKB spec for the benchmark to run.
+    modulename: The name of the python module to run.
+    job_arguments: The additional job arguments provided for the run.
+
+  Returns:
+    cmd: Array containg the built command.
+  """
+
+  cmd = []
+
+  python_executable = FLAGS.python_binary
+  if not vm_util.ExecutableOnPath(python_executable):
+    raise errors.Setup.MissingExecutableError(
+        'Could not find required executable "%s"' % python_executable)
+  cmd.append(python_executable)
+
+  cmd.append('setup.py')
+  cmd.append('nosetests')
+  cmd.append('--tests={}'.format(modulename))
+  cmd.append('--attr={}'.format(FLAGS.beam_python_attr))
+
+  beam_args = job_arguments if job_arguments else []
+
+  if benchmark_spec.service_type == dpb_service.DATAFLOW:
+    python_binary = _FindFiles(os.path.join(_GetBeamDir(),
+                                            'sdks/python/target'),
+                               'apache-beam*.tar.gz')
+    if len(python_binary) == 0:
+      raise RuntimeError('No python binary is found')
+
+    beam_args.append('--runner=TestDataflowRunner')
+    beam_args.append('--sdk_location={}'.format(python_binary[0]))
+
+  cmd.append('--test-pipeline-options='
+             '{}'.format(' '.join(beam_args)))
+
+  return cmd
+
+
+def _GetBeamDir():
   # TODO: This is temporary, find a better way.
-  beam_dir = FLAGS.beam_location if FLAGS.beam_location else os.path.join(
+  return FLAGS.beam_location if FLAGS.beam_location else os.path.join(
       vm_util.GetTempDir(), 'beam')
-  return cmd, beam_dir
+
+
+def _FindFiles(base_path, pattern):
```

Comment:
Do you think this can be used ? 
https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/4db490efb48ebb33584ac159580c271f5e1d37f5/perfkitbenchmarker/data/__init__.py#L103